src/tools/components/InputParameters.tsx

```diff
@@ -8,43 +7,73 @@ import { UseDataAPIState } from '../../shared/hooks/useDataApi';
 
 import { PublicServerParameters } from '../types/toolsServerParameters';
 import { JobTypes } from '../types/toolsJobTypes';
+import { FinishedJob } from '../types/toolsJob';
 
 type InputParametersProps = {
   id: string;
-  inputParamsData: Partial<UseDataAPIState<PublicServerParameters[JobTypes]>>;
+  // No public endpoint to expose this for peptide search, so for now replace
+  // with a "possible" job object in the case it's the same user that created it
+  inputParamsData:
+    | Partial<UseDataAPIState<PublicServerParameters[JobTypes]>>
+    | FinishedJob<JobTypes.PEPTIDE_SEARCH>
+    | null;
   jobType: JobTypes;
 };
 
-const InputParameters: FC<InputParametersProps> = ({
+const fieldsToHide = new Set(['redirectURL']);
+
+const InputParameters = ({
   id,
   inputParamsData,
   jobType,
-}) => {
-  const { loading, data, error, status } = inputParamsData;
+}: InputParametersProps) => {
+  if (inputParamsData && !('type' in inputParamsData)) {
+    if (inputParamsData.error || !inputParamsData.data) {
+      return <ErrorHandler status={inputParamsData.status} />;
+    }
+  }
 
-  if (error || !data) {
-    return <ErrorHandler status={status} />;
+  if (!inputParamsData) {
+    if (jobType === JobTypes.PEPTIDE_SEARCH) {
+      return (
+        <section>
+          <p>
+            Unable to retrieve the input values for the job with ID{' '}
+            <code>{id}</code>.<br />
+            It looks like this job hasn&apos;t been submitted from this browser.
+            Please ask the person that submitted this job for this information.
+          </p>
+        </section>
+      );
+    }
+    return <ErrorHandler />;
   }
 
+  const inputParameters =
+    'parameters' in inputParamsData
+      ? inputParamsData.parameters
+      : inputParamsData.data;
+
   return (
     <>
-      <ResubmitButton
-        inputParamsData={inputParamsData.data}
-        jobType={jobType}
-      />
+      {inputParameters && (
+        <ResubmitButton inputParamsData={inputParameters} jobType={jobType} />
+      )}
       <section>
         <p>
-          The job with UUID <code>{id}</code> has been submitted with these raw
+          The job with ID <code>{id}</code> has been submitted with these raw
           input values:
         </p>
-        {loading ? (
+        {'loading' in inputParamsData && inputParamsData.loading ? (
           <Loader />
         ) : (
           <InfoList
-            infoData={Object.entries(data).map(([key, value]) => ({
-              title: key,
-              content: <CodeBlock lightMode>{`${value}`}</CodeBlock>,
-            }))}
+            infoData={Object.entries(inputParameters || {})
+              .filter(([key, value]) => !fieldsToHide.has(key) && value)
+              .map(([key, value]) => ({
+                title: key,
+                content: <CodeBlock lightMode>{`${value}`}</CodeBlock>,
```

Comment:
<img width="2160" alt="Screenshot 2022-03-11 at 10 57 52" src="https://user-images.githubusercontent.com/4357962/157854860-51150f32-bc6a-4b9d-9858-33c2eeb81c49.png">

Would it be possible to wrap long sequences?

Reply:
Quickly touched a bit this area in order to try to solve this but didn't manage to do so quickly. It feels like I'd have to fight against some patterns within the InfoList, DecoredListItem, and CodeBlock components in Franklin.
For now I just made it so that it scrolls the sequence section if it's too big and not the whole page.
Feel free to have another go at it if you want.